### PR TITLE
fixes #27806 - use PG by default in docker-compose

### PR DESCRIPTION
**Pull request by ohadlevy**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -2,30 +2,30 @@ version: '3.4'
...
+      - PGDATA=/var/lib/postgresql/data/pgdata
     hostname: db.example.com
-    image: centos/mariadb-102-centos7
+    image: postgres:10
     ports:
-      - 3306
+      - 5432
     restart: always
     healthcheck:
       test: ["CMD-SHELL", "nc -z 127.0.0.1 3306 || exit 1"]
```
Comment (ekohl) [inline docker-compose.yml]: Postgres is port 5432
```suggestion
       test: ["CMD-SHELL", "nc -z 127.0.0.1 5432 || exit 1"]
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (ohadlevy) [reply docker-compose.yml]: thanks, updated.

**Review by ekohl — APPROVED** [review]
Untested, but looks correct to me

File: docker-compose.yml [hunk 2]
```diff
@@ -2,30 +2,30 @@ version: '3.4'
...
   db:
     environment:
-      - MYSQL_USER=foreman
-      - MYSQL_PASSWORD=foreman
-      - MYSQL_DATABASE=foreman
-      - MYSQL_DATADIR_ACTION=upgrade-auto
+      - POSTGRES_USER=foreman
+      - POSTGRES_PASSWORD=foreman
+      - POSTGRES_DATABASE=foreman
+      - PGDATA=/var/lib/postgresql/data/pgdata
```
Comment (lzap) [inline docker-compose.yml:8]: This looks like the standard path, is this here just for better visibility?

**Review by lzap — APPROVED** [review]
LGTM as well.

File: docker-compose.yml (hunk 2 above)
Comment (ohadlevy) [reply docker-compose.yml:8]: it actually adds another subdir (pgdata under data) this ensures that the mounted volume is not within the container (see https://hub.docker.com/_/postgres) it follow a similar issue I had on the openshift implementation (at https://github.com/ohadlevy/foreman-kube)

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Thanks, @ohadlevy.

**Pull request merged**